test/test_dataloader.py

```diff
@@ -2605,6 +2605,41 @@ def test_ind_worker_queue(self):
                 self._run_ind_worker_queue_test(batch_size=batch_size, num_workers=num_workers + 1)
 
 
+class TestMultiEpochDataset(IterableDataset):
+    def __init__(self, length):
+        self.length = length
+
+    def __iter__(self):
+        worker_info = torch.utils.data.get_worker_info()
+        assert worker_info is not None
+        worker_id = worker_info.id
+        assert self.length % worker_info.num_workers == 0
+        for idx in range(self.length // worker_info.num_workers):
+            if worker_id == 0:
+                time.sleep(0.001)
+            yield worker_id
+
+    def __len__(self):
+        return self.length
+
+
+class TestMultiEpochDeterminism(TestCase):
+    def test_multi_epochs_deter(self):
+        num_workers = 2
+        batch_size = 10
+        num_epochs = 3
+
+        dataset = TestMultiEpochDataset(batch_size*num_workers)
+
+        loader = DataLoader(
+            dataset, batch_size=batch_size, num_workers=num_workers,
+            shuffle=False,  persistent_workers=True, timeout=5
+        )
```

Comment:
Could you please check both `True` and `False` for `persistent_workers`?

Reply:
it works with False.  it's a good check since we remove the iterator intialization in __init__

Reply:
added test for false